src/Compilers/CSharp/Test/Semantic/Semantics/PropertyFieldKeywordTests.cs

```diff
@@ -0,0 +1,1186 @@
+");
+            Assert.Equal(0, accessorBindingData.NumberOfPerformedAccessorBinding);
+        }
+
+        [Fact(Skip = "PROTOTYPE(semi-auto-props): Mixed scenarios are not yet supported.")]
+        public void TestPrefixedWithAt()
+        {
+            var comp = CreateCompilation(@"
+public class C
+{
+    public string P { get; set => @field = value; }
+}
+");
+            var accessorBindingData = new SourcePropertySymbolBase.AccessorBindingData();
+            comp.TestOnlyCompilationData = accessorBindingData;
+            comp.VerifyDiagnostics(
+                // (4,35): error CS0103: The name 'field' does not exist in the current context
+                //     public string P { get; set => @field = value; }
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "@field").WithArguments("field").WithLocation(4, 35));
+            Assert.Equal(0, accessorBindingData.NumberOfPerformedAccessorBinding);
+        }
+
+        [Fact]
+        public void TestPrefixedWithAt2()
+        {
+            var comp = CreateCompilation(@"
+public class C
+{
+    public string P { get => @field; }
+}
+");
+            var accessorBindingData = new SourcePropertySymbolBase.AccessorBindingData();
+            comp.TestOnlyCompilationData = accessorBindingData;
+            comp.VerifyDiagnostics(
+                // (4,30): error CS0103: The name 'field' does not exist in the current context
+                //     public string P { get => @field; }
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "@field").WithArguments("field").WithLocation(4, 30));
+            Assert.Equal(0, accessorBindingData.NumberOfPerformedAccessorBinding);
+        }
+
+        [Fact]
+        public void TestHasFieldMemberInScope()
+        {
+            var comp = CreateCompilation(@"
+public class B
+{
+    protected string field;
+}
+public class C : B
+{
+    public string P { get => field; }
+}
+");
+            var accessorBindingData = new SourcePropertySymbolBase.AccessorBindingData();
+            comp.TestOnlyCompilationData = accessorBindingData;
+            VerifyTypeIL(comp, "C", @"
+.class public auto ansi beforefieldinit C
+	extends B
+{
+	// Methods
+	.method public hidebysig specialname 
+		instance string get_P () cil managed 
+	{
+		// Method begins at RVA 0x2058
+		// Code size 7 (0x7)
+		.maxstack 8
+		IL_0000: ldarg.0
+		IL_0001: ldfld string B::'field'
+		IL_0006: ret
+	} // end of method C::get_P
+	.method public hidebysig specialname rtspecialname 
+		instance void .ctor () cil managed 
+	{
+		// Method begins at RVA 0x2060
+		// Code size 7 (0x7)
+		.maxstack 8
+		IL_0000: ldarg.0
+		IL_0001: call instance void B::.ctor()
+		IL_0006: ret
+	} // end of method C::.ctor
+	// Properties
+	.property instance string P()
+	{
+		.get instance string C::get_P()
+	}
+} // end of class C
+");
+            Assert.Equal(1, accessorBindingData.NumberOfPerformedAccessorBinding); // PROTOTYPE(semi-auto-props): Is this expected?
```

Comment:
>// PROTOTYPE(semi-auto-props): Is this expected? [](http://example.com/codeflow?start=83&length=48)

No, this looks unexpected to me. I think we should address this in this PR. Here is what I think should work. In ```MethodCompiler.CompileNamedType``` after the loop that compiles all members, we can do another loop through properties and force assign ```_lazyBackingFieldSymbol``` to null if it is still in an unknown state. Obviously, this should be done through a special API on the property symbol.

Reply:
Fixed